File: Frontend/lib/domain/hospital/entity/hospital.dart

import 'package:special_tests/domain/special_tests/entity/special_tests_model.dart';

class Hospital {
  String hospId;
  String name;
  String description;
  double rating;
  int ratedClients;
  String address;
  String image;
  int requestedTimes;
  List<SpecialTest> specialTest;

  static final List<String> hospitalFields = [
    'hosp_id',
    'name',
    'description',
    'rating',
    'rated_clients',
    'address',
    'image',
    'requestedTimes'
  ];

  Hospital({
    required this.hospId,
    required this.name,
    required this.description,
    required this.rating,
    required this.ratedClients,
    required this.address,
    required this.image,
    required this.requestedTimes,
    required this.specialTest,
  });

  factory Hospital.fromJson(Map<String, dynamic> json) {
    return Hospital(
        hospId: json['hosp_id'],
        name: json['name'],
        description: json['description'],
        rating: json['rating'],
        ratedClients: json['ratedClients'],
        address: json['address'],
        image: json['image'],
        requestedTimes: json['requestedTimes'],
        specialTest: json['specialTests']);
  }

  Map<String, dynamic> toJson() {
    Map<String, dynamic> json = {};
    json['hosp_id'] = hospId;
    json['name'] = name;
    json['description'] = description;
    json['rating'] = rating;
    json['ratedClients'] = ratedClients;
    json['address'] = address;
    json['image'] = image;
    json['requestedTimes'] = requestedTimes;
    return json;
  }

  @override
  String toString() {
    return "client id $hospId, name $name";
  }
}

File: Backend/src/main/java/com/mobapp/SpecialTest/models/Hospital.java
package com.mobapp.SpecialTest.models;

import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.JoinTable;
import javax.persistence.ManyToMany;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;
import javax.persistence.Table;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonProperty.Access;

import lombok.Data;


  
@Entity
@Table(name = "Hospital") @Data
public class Hospital{
        
    @Id
    @GeneratedValue(strategy =  GenerationType.IDENTITY)
    private long hosp_id;
    private String name;
    private String description;
    private Float rating;
    private int ratedClients;
    private String address;
    private int requestedTimes;

    @ManyToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JoinTable(name = "hospital_test", joinColumns = @JoinColumn(name = "hosp_id"),  inverseJoinColumns = @JoinColumn(name = "test_id"))
    private List<SpecialTest> specialTest;

    @OneToMany(fetch = FetchType.LAZY)
    @JsonProperty(access = Access.WRITE_ONLY)
    private List<Request> request;
    
    // @OneToOne(cascade = CascadeType.ALL)
    // @JoinColumn(name = "hosp_id", referencedColumnName = "user_id")
    @OneToOne
    private User user;

    // @JsonProperty(access = Access.WRITE_ONLY)
    // public Long getClient_id(){
    //     return this.hosp_id;
    // }
    
}
    



